go/ct/st/memory.go

```diff
@@ -111,13 +125,20 @@ func (m *Memory) ExpansionCosts(offset_u256, size_u256 U256) (memCost tosca.Gas,
 		memCost = 0
 		return
 	}
+
+	if newSize > maxMemoryExpansionSize {
+		memCost = MaxGas
```

Comment:
In the other cases where we have saturated memory costs we return `math.MaxInt64`. Here, it returns `MaxGas`. This is not consistent. I think this here should be `math.MaxInt64` as well.

Note that `MaxGas` is not `tosca.Gas(math.MaxInt64)`.

Reply:
I am aware of the difference, I thought using `MaxGas` was better since it is smaller than `MaxInt64` but this is intentional, as described in #293. I think we should keep it consistent, yes, but that given that there is an existing issue about it, I thought sticking with `MaxGas` was the safer path. Should we remove the issue and remove all uses of `MaxGas` or stick with it and change other uses of `MaxInt64` to `MaxGas` ?